### PR TITLE
[ARM CPU] Avg Pooling, ROI Pooling fix for fp16 precision

### PR DESCRIPTION
**Pull request by allnes**
### Details:
 - *item1*
 - *...*

### Tickets:
 - CVS-123064
 - CVS-123019 (partially)

### CODE REVIEW

**Comment by vurusovs** [comment]
@allnes please, take into account ROIPooling tests are skipped in scope of https://github.com/openvinotoolkit/openvino/pull/20552 (just merged)

File: src/plugins/intel_cpu/src/nodes/roi_pooling.cpp [hunk 1]
```diff
@@ -826,7 +826,8 @@ std::shared_ptr<ROIPooling::ROIPoolingExecutor> ROIPooling::ROIPoolingExecutor::
 
     OV_SWITCH(intel_cpu, ROIPoolingExecutorCreation, ctx, jpp.src_prc,
               OV_CASE(Precision::FP32, float),
-              OV_CASE(Precision::BF16, bfloat16_t))
+              OV_CASE(Precision::BF16, bfloat16_t),
+              OV_CASE(Precision::FP16, float16_t))
```
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/nodes/roi_pooling.cpp:830]: I think x86 jit executor does not support fp16 precision
I highly recommend to double check the fp16 changes are working, using SPR host or sde emulator.

Comment (allnes) [reply src/plugins/intel_cpu/src/nodes/roi_pooling.cpp:831]: @EgorDuplensky corrected

File: src/plugins/intel_cpu/src/nodes/roi_pooling.cpp [hunk 2]
```diff
@@ -453,6 +453,10 @@ void ROIPooling::initSupportedPrimitiveDescriptors() {
         impl_type = impl_desc_type::ref;
     }
 
+    if (impl_type != impl_desc_type::ref && refParams.src_prc == Precision::FP16) {
```
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/src/nodes/roi_pooling.cpp:456]: let's have src_prc assignment in one place of the function

Comment (allnes) [reply src/plugins/intel_cpu/src/nodes/roi_pooling.cpp:456]: @EgorDuplensky corrected

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**